roles/ceph-osd/tasks/bcache.yml

```diff
@@ -48,6 +48,15 @@
   when: journal_partitions.rc != 0 and
         bcache_partition.rc != 0
 
+- name: initialize partition_connector flag
+  set_fact:
+    partition_connector: ''
+
+- name: set partition_connector flag for nvme
+  set_fact:
+    partition_connector: 'p'
+  when: '"nvme" in {{ ceph.bcache_ssd_device }}'
```

Comment:
ceph.bcache_ssd_device is not a list
change this to: ceph.bcache_ssd_device|search('nvme')